src/Button.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { memo, forwardRef } from "react";
+// import { symToStr } from "tsafe/symToStr";
+import { fr } from "./lib";
+import { cx } from "./lib/tools/cx";
+
+// We make users import dsfr.css, so we don't need to import the scoped CSS
+// but in the future if we have a complete component coverage it
+// we could stop requiring users to import the hole CSS and only import on a
+// per component basis.
+import "./dsfr/component/button/button.css";
+
+// This is just an example, we should get types from .fr-icon-* list
+// const icons = ["fr-icon-checkbox-circle-line", "fr-icon-account-circle-fill"] as const;
+// type IconType = typeof icons[number];
+
+type IconType = `fr-icon-${string}`;
+
+type ButtonIcon = {
+    name: IconType;
+    position?: "left" | "right";
+};
+
+type ButtonCommonProps = {
+    label: string;
+    icon?: ButtonIcon;
+    priority?: "secondary" | "tertiary";
+    className?: string;
+    size?: ButtonProps.Size;
+};
+
+export type ButtonProps = ButtonCommonProps & (ButtonProps.Anchor | ButtonProps.Button);
+
+export namespace ButtonProps {
+    export type Size = "sm" | "lg";
+    export type Anchor = {
+        href: string | null;
+        target?: React.HTMLAttributeAnchorTarget;
+        disabled?: never;
+        type?: never;
+        onClick?: never;
+    };
+    export type Button = {
+        onClick?: React.MouseEventHandler<HTMLButtonElement>;
+        disabled?: boolean;
+        type?: "button" | "submit" | "reset";
+        href?: never;
+        target?: never;
+    };
+}
+
+export const Button = memo(
+    forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(props => {
+        const { icon, priority, className, size, label } = props;
+
+        const buttonClassName = cx(
+            fr.cx("fr-btn"),
+            priority && fr.cx(`fr-btn--${priority}`),
+            size && fr.cx(`fr-btn--${size}`),
+            icon && cx(fr.cx(icon.name), icon.position && fr.cx(`fr-btn--icon-${icon.position}`)),
+            className
+        );
+        const Component =
+            "href" in props ? (
+                <a
+                    className={buttonClassName}
+                    href={props.href ?? undefined}
+                    target={props.target || "_self"}
+                >
+                    {label}
+                </a>
```

Comment:
We can's use `<a />` directly.   
It was ok for skip links but in any other case we have to use the `Link` component that is provided to us and use the type `RegistredLinkProps` (With `react-router` for example there is no `href` prop but a `to`).    

See: https://react-dsfr.etalab.studio/routing
and for example: https://github.com/codegouvfr/react-dsfr/blob/c1815499214e0f7be66c05d6b82fa62f589c286b/src/Header/Header.tsx#L124

Reply:
There's some things to consider: 
- for accessibility/semantic sakes, if an we make references to inner sections / pages using links, we should use `<a>` with an `href`. <button> shouldn't be rendered in the DOM in those cases
- how should we handle a Button component that points to an external document?
- to be more generic, I think of this component only as a UI Button component, and we should be able to use it the way we can't, as long as it follows DSFR, and ARA guidelines.